lib/bolt/inventory.rb

```diff
@@ -434,5 +434,17 @@ def invalidate_group_cache!(target)
       @target_plugin_hooks.delete(target.name)
     end
     private :invalidate_group_cache!
+
+    # Logs deprecation warning for Target.new(<uri>, <options>)
+    def target_deprecation
```

Comment:
The inventory here is actually just keeping track of whether we've already printed a deprecation message or not. I think it makes more sense to have the deprecation messages be defined where they're called, since the message is distinct for each case, and to track if they've been printed as class variables on the inventory class. `args_deprecation_issued` and `hash_deprecation_issued` or something, that are set to false in the inventory initialization and are in `attr_accessor` to be read and reset in the target class.